File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode: public rclcpp::Node
{
public:
    RobotNewsStationNode(): Node("robot_news_station"), m_robotName("R2D2")
    {
        m_publisher = this->create_publisher<example_interfaces::msg::String>("robot_news",10);


        m_timer = this->create_wall_timer(std::chrono::milliseconds(5000),std::bind(&RobotNewsStationNode
        ::callback_robotNews,this));

        RCLCPP_INFO(this->get_logger(),"Robot news Station has been started. ");  
    
    }

private:

    void callback_robotNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hi this is ") + m_robotName + " from robot news station. ";
        m_publisher->publish(msg);
    }
    std::string m_robotName;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr m_publisher;
    rclcpp::TimerBase::SharedPtr m_timer;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class RobotNewsStationNode(Node):

    def __init__(self):
        super().__init__("robot_news_station")

        

        self.declare_parameter("robot_name","C3P0")
        self.declare_parameter("publish_frequency",1.0)

        self.robot_name_ = self.get_parameter("robot_name").value

        self.publish_frequency = self.get_parameter("publish_frequency").value

        self.publisher_ = self.create_publisher(String,"robot_news",10)

        #self.parameters = self.declare_parameter("test124")

        self.timer_ = self.create_timer(1.0 / self.publish_frequency, self.publish_news)

        self.get_logger().info("Robot News Stations has started")
        
    def publish_news(self):
        msg = String()
        msg.data = "Hello, this  is " + self.robot_name_ + " from robot news station "

        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
    
